### PR TITLE
Unescape the HTML-escaped HTML.

### PR DESCRIPTION
**Pull request by brycied00d**
Matrix uses JSON to pass messages, and the Matrix API functions are perfectly capable of handling the required escaping. HTML needs no escaping.

Before this fix, what was sent and ultimately seen by other clients:  
`This is &quot;quoted&quot; 1234`  
With this fix: `This is "quoted" 1234`

### CODE REVIEW

**Comment by richvdh** [comment]
Looks plausible, but looks like it leaks memory - you aren't freeing the returned string.

Please could you sign off the PR or commit as per CONTRIBUTING.rst.

**Comment by brycied00d** [comment]
Done and done, @richvdh .

File: matrix-room.c [hunk 1]
```diff
@@ -1020,10 +1020,8 @@ void matrix_room_send_message(PurpleConversation *conv, const gchar *message)
         return;
```
Comment (richvdh) [inline matrix-room.c]: still leaks memory here.

In fact it looks like we shouldn't be doing the unescaping on this path?

File: matrix-room.c [hunk 2]
```diff
@@ -1020,10 +1020,8 @@ void matrix_room_send_message(PurpleConversation *conv, const gchar *message)
...
     }
 
-    if (!strncmp(message, "/me ", 4)) {
+    if (purple_message_meify(message_to_send, -1))
         type_string = "m.emote";
-        message_to_send = message + 4;
-    }
```
Comment (richvdh) [inline matrix-room.c]: btw, generally I prefer to keep braces on one-line if bodies, but certainly I'll accept that the code is not a model of consistency on this front.

File: matrix-room.c (hunk 1 above)
Comment (brycied00d) [inline matrix-room.c]: > In fact it looks like we shouldn't be doing the unescaping on this path?

We shouldn't, and we aren't. The image itself requires no unescaping. The unescaped form of the message is only needed for the `m.text` message body, so moving `message_to_send` assignment after the image-handling will address this.

**Comment by brycied00d** [comment]
I've made the changes requested, and squashed this into a single commit for simplicity.

**Comment by richvdh** [comment]
Looks great. Thanks very much!

**Comment by brycied00d** [comment]
Happy to help - my friends and I got really tired of seeing `&quot;`

Next up: support for `org.matrix.custom.html` (in a separate PR, of course.)

**Pull request merged**